routes/profileRoute.js

```diff
@@ -0,0 +1,143 @@
+const express = require('express');
+const router = require('express').Router();
+const cookieParser = require('cookie-parser');
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcryptjs');
+const User = require('../models/User');
+const Team = require('../models/Team');
+const Template = require('../models/Template');
+
+const maxAge = 30 * 24 * 60 * 60;
+const createToken = (id) => {
+   return jwt.sign({
+       id
+   }, process.env.TokenSecret, {
+       expiresIn: maxAge
+   });
+}
+
+router.post('/addtemplate', async (req, res) => {
+    const token = req.cookies.jwt;
+    const decoded = jwt.verify(token, process.env.TokenSecret);
+    var userId = decoded.id;
+    const templateExists = await Template.findOne({
+        userId: userId
+    });
+    if (templateExists === null){
+        const template = new Template({
```

Comment:
Called /addtemplate with this body after logging in:
{
    "workoutName": "Home workout",
    "exercises": ["Raises", "Lunges"],
    "note": ["Hello"]
}
and I'm getting the 'Failed to create template' log. Did it work for you? I don't think we can create a template with just the user id. i think we need to add a default to the schema or pass null